standard/tier/tier_utils.lua

```diff
@@ -0,0 +1,177 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Tier/Utils
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+-- module intended to be moved to `Module:Tier` after the old ones usage has been eliminated
+
+local HiddenSort = require('Module:HiddenSort')
+local Logic = require('Module:Logic')
+local Page = require('Module:Page')
+local String = require('Module:StringUtils')
+
+local TierData = mw.loadData('Module:Tier/Data')
+
+local NON_BREAKING_SPACE = '&nbsp;'
+
+local Tier = {}
+
+--- Converts input to standardized identifier format
+---@param input string|integer|nil
+---@return string|integer|nil
+function Tier.toIdentifier(input)
+	if String.isEmpty(input) then
+		return ''
+	end
+
+	if Logic.isNumeric(input) then
+		return tonumber(input)
+	end
+
+	return string.lower(input):gsub(' ', '')
+end
+
+--- Retrieves the raw tier/tierType data for a given input
+---@param input string|integer|nil
+---@param mode 'tiers'|'tiertypes'
+---@return table?
+function Tier.raw(input, mode)
+	return (TierData[mode] or {})[input]
+end
+
+--- Converts input to storage value for tier/tierType
+---@param input string|integer|nil
+---@param mode 'tiers'|'tiertypes'
+---@return boolean
+function Tier.isValid(input, mode)
+	return String.isEmpty(input) or Tier.raw(input, mode) ~= nil
+end
+
+--- Converts input to (storage) value for tier/tierType
+---@param input string|integer|nil
+---@param mode 'tiers'|'tiertypes'
+---@return string|integer|nil
+function Tier.toValue(input, mode)
+	return (Tier.raw(input, mode) or {}).value
+end
+
+--- Converts input to display name for tier/tierType
+---@param input string|integer|nil
+---@param mode 'tiers'|'tiertypes'
+---@return string?
+function Tier.toName(input, mode)
+	return (Tier.raw(input, mode) or {}).name
+end
+
+--- Converts input to short name for tier/tierType
+---@param input string|integer|nil
+---@param mode 'tiers'|'tiertypes'
+---@return string?
+function Tier.toShortName(input, mode)
+	return (Tier.raw(input, mode) or {}).short
+end
+
+--- Converts input to (portal) page for tier/tierType
+---@param input string|integer|nil
+---@param mode 'tiers'|'tiertypes'
+---@return string?
+function Tier.toLink(input, mode)
+	return (Tier.raw(input, mode) or {}).link
+end
+
+--- Converts input to a tier/tierType category
+---@param input string|integer|nil
+---@param mode 'tiers'|'tiertypes'
+---@return string?
+function Tier.toCategory(input, mode)
+	return (Tier.raw(input, mode) or {}).category
+end
```

Comment:
`toLink` and `toCategory` are used rarely enough that I think the accessor functions aren't needed. `Tier.getData(tier).link` is sufficient.

If you do keep them then consider renaming to `getLink/getCategory` because they're not two-way conversions but rather one-way field lookups. (there shouldn't be a need to convert a link/category back into a tier)

Reply:
to category usecase would be infobox league mostly
to link might get kicked

Reply:
toLink kicked
imo toCategory should stay for e.g. infobox league

Reply:
ok. toCategory will be used more than once I gather?

Reply:
yeah